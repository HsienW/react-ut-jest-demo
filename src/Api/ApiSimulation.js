export const successData = {
    state: 'success',
    data: [
        {
            id: 1,
            name: 'Apple'
        },
        {
            id: 2,
            name: 'Banana'
        },
        {
            id: 3,
            name: 'Tomato'
        }
    ]
};

export const failData = {
    state: 'fail',
    data: []
};

export const ApiSimulation = () => {
    return new Promise((resolve, reject) => {
        const success = true;
        if (success) {
            setTimeout(() => {
                resolve(successData);
            }, 500);
        } else {
            reject(failData);
        }
    });
};