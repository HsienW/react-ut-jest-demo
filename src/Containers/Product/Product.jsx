import React, {Component} from 'react';
import {ApiSimulation} from '../../Api/ApiSimulation';
import {ProductList} from '../../Components/ProductList/ProductList';

// import PropTypes from 'prop-types';


export class Product extends Component {

    state = {
        getListDataState: '',
        listData: []
    };

    componentDidMount() {
        ApiSimulation()
            .then((respond) => {
                console.log('tttttttttttttttttttt');
                console.log(respond);
                this.setState({
                    getListDataState: respond.state,
                    listData: respond.data
                });
            })
            .catch((error) => {
                this.setState({
                    getListDataState: error.state,
                    listData: error.data
                });
            });
    }

    render() {
        return (
            <div>
                <h1>Product Page</h1>
                <ProductList listData={this.state.listData}/>
            </div>
        );
    }
}

// Product.propTypes = {
// };
