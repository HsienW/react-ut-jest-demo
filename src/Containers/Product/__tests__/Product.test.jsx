import React from 'react';
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Product} from '../Product';
import {ApiSimulation, successData} from '../../../Api/ApiSimulation';

Enzyme.configure({adapter: new Adapter()});


describe('Product container', () => {
    const wrapper = shallow(<Product/>);

    it('should created container', () => {
        expect(wrapper.exists()).toEqual(true);
    });

    it('should created snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});

describe('ProductList state', () => {
    it('should mock componentDidMount call api', async () => {
        const wrapper = mount(<Product />);
        await ApiSimulation();
        wrapper.update();
        expect(wrapper.state('getListDataState')).toEqual('success');
        expect(wrapper.state('originalListData')).toEqual(successData.data);
        expect(wrapper.state('displayListData')).toEqual(successData.data);
    });
});
