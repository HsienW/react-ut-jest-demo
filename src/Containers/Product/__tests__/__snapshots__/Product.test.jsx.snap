// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App container should created snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Product />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Product Page
        </h1>,
        <ProductList
          listData={Array []}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Product Page",
        },
        "ref": null,
        "rendered": "Product Page",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "listData": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Product Page
          </h1>,
          <ProductList
            listData={Array []}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Product Page",
          },
          "ref": null,
          "rendered": "Product Page",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "listData": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Product container should created snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Product />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Product Page
        </h1>,
        <ProductList
          listData={Array []}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Product Page",
        },
        "ref": null,
        "rendered": "Product Page",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "listData": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Product Page
          </h1>,
          <ProductList
            listData={Array []}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Product Page",
          },
          "ref": null,
          "rendered": "Product Page",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "listData": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
